packages/flutter/lib/src/material/drawer.dart

```diff
@@ -217,7 +221,18 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
   }
 
   void _move(DragUpdateDetails details) {
-    final double delta = details.primaryDelta / _width;
+    if (_controller.isDismissed) {
+      final ModalRoute<dynamic> parentRoute = ModalRoute.of(context);
+      final bool canPop = parentRoute?.canPop ?? false;
+      if(canPop) {
+        Navigator.of(context).pop();
+        return;
+      }
+    }
+    double delta = details.primaryDelta / _width;
+    if(widget.type == DrawerAlignment.END) {
```

Comment:
Please use a `switch` statement for enums.  That way the analyzer will point us to this line of code if we ever add more alignments (e.g., bottom)

Reply:
Fixed